internal/util/idlocker_test.go

```diff
@@ -50,3 +50,80 @@ func TestIDLocker(t *testing.T) {
 			true, ok)
 	}
 }
+
+func TestOperationLocks(t *testing.T) {
+	volumeID := "test-vol"
+	lock := NewOperationLock()
+	err := lock.GetCloneLock(volumeID)
+
+	if err != nil {
+		t.Errorf("failed to acquire clone lock for %s %s", volumeID, err)
+	}
+	err = lock.GetDeleteLock(volumeID)
+	if err == nil {
+		t.Errorf("expected to fail for GetDeleteLock for %s", volumeID)
+	}
+
+	err = lock.GetExpandLock(volumeID)
+	if err == nil {
+		t.Errorf("expected to fail for GetExpandLock for %s", volumeID)
+	}
+	lock.ReleaseCloneLock(volumeID)
+
+	// Get multiple clone operation
+	err = lock.GetCloneLock(volumeID)
+	if err != nil {
+		t.Errorf("failed to acquire clone lock for %s %s", volumeID, err)
+	}
+	err = lock.GetCloneLock(volumeID)
+	if err != nil {
+		t.Errorf("failed to acquire clone lock for %s %s", volumeID, err)
+	}
+	err = lock.GetCloneLock(volumeID)
+	if err != nil {
+		t.Errorf("failed to acquire clone lock for %s %s", volumeID, err)
+	}
+	// release all clone locks
+	lock.ReleaseCloneLock(volumeID)
+	lock.ReleaseCloneLock(volumeID)
+	lock.ReleaseCloneLock(volumeID)
```

Comment:
what happens if there are more releases than acquires?

Reply:
its safer no op nothing will happen as the key won't be present to delete it

Reply:
ok